cmd/dep/gopath_scanner.go

```diff
@@ -119,17 +120,32 @@ func (g *gopathScanner) overlay(rootM *dep.Manifest, rootL *dep.Lock) {
 	}
 
 	// Identify projects whose version is unknown and will have to be solved for
-	var unlockedProjects []string
+	var missing []string    // all project roots missing from GOPATH
+	var missingVCS []string // all project roots missing VCS information
 	for pr := range g.pd.notondisk {
 		if _, isLocked := lockedProjects[pr]; isLocked {
 			continue
 		}
-		unlockedProjects = append(unlockedProjects, string(pr))
+		if g.pd.invalidSVC[pr] {
+			missingVCS = append(missingVCS, string(pr))
+		} else {
+			missing = append(missing, string(pr))
+		}
+	}
+
+	missingStr := ""
+	missingVCSStr := ""
+	if len(missing) > 0 {
+		missingStr = fmt.Sprintf("Following dependencies were not found in GOPATH:\n  %s\n",
+			strings.Join(missing, "\n  "))
+	}
+	if len(missingVCS) > 0 {
+		missingVCSStr = fmt.Sprintf("Following dependencies found in GOPATH were missing VCS information:\n  %s\n",
+			strings.Join(missingVCS, "\n  "))
 	}
-	if len(unlockedProjects) > 0 {
-		g.ctx.Err.Printf("Following dependencies were not found in GOPATH. "+
-			"Dep will use the most recent versions of these projects.\n  %s",
-			strings.Join(unlockedProjects, "\n  "))
+	if len(missingVCS)+len(missing) > 0 {
+		g.ctx.Err.Printf("%s%sDep will use the most recent versions of these projects."+
+			" (Dep will not work if a remote source for these projects doesn't exist)\n\n", missingStr, missingVCSStr)
```

Comment:
> Dep will not work if a remote source for these projects doesn't exist

That is in reference to the missing vcs info, correct? Let's move that closer to the message.

```
The following dependencies were not found in GOPATH:
  github.com/doesnt/exist
  github.com/alsodoesnt/exist

The following dependencies found in GOPATH were missing VCS information (a remote source is required):
  github.com/user/foo
  github.com/user/bar

The most recent version of these projects will be used.
```

I corrected some existing grammar errors in the above message, if you don't mind including those fixes as well, that would be helpful!
* "Following dependencies" -> "The following dependencies"
* "Dep will use" -> "The most recent version of these projects will be used"

Reply:
Perfect. I'll do that. Displaying it that way does look better.